src/async_runtime.rs

```diff
@@ -0,0 +1,109 @@
+use crate::errors::InitializationError;
+
+use core::future::Future;
+use core::time::Duration;
+use tokio::runtime::{Builder, Runtime};
+use tokio::task::JoinHandle;
+use tokio::time::{interval, MissedTickBehavior};
+
+pub struct AsyncRuntime {
+    rt: Runtime,
+}
+
+impl AsyncRuntime {
+    pub fn new() -> Result<Self, InitializationError> {
+        let rt = Builder::new_multi_thread()
+            .worker_threads(4)
+            .thread_name("3l-asyn-runtime")
+            .enable_time()
+            .build()
+            .map_err(|e| InitializationError::AsyncRuntime {
+                message: e.to_string(),
+            })?;
+        Ok(Self { rt })
+    }
+
+    pub fn spawn<F>(&self, future: F) -> JoinHandle<F::Output>
+    where
+        F: Future + Send + 'static,
+        F::Output: Send + 'static,
+    {
+        self.rt.spawn(future)
+    }
+
+    pub fn spawn_regularly<Func, F>(&self, duration: Duration, func: Func) -> JoinHandle<()>
```

Comment:
I think this name is a bit confusing. It makes it seem like `spawn` will be called regularly but it's called only once. Not sure about the following suggestion either, but I think it's an improvement.

```suggestion
    pub fn spawn_repeating_task<Func, F>(&self, duration: Duration, func: Func) -> JoinHandle<()>
```